Develop

yunustuncbilek opened the pull request:
Some of new features: 
1. Most parameters are coming from the launch file
2. New message types `FreeObject`, `Screen` etc
3. Tracking the gaze pointing at the robot using Line-Point Distance formula, or any free object (where free object is anything not on the screen)
4. Topic connecting target_detection and find_gazepoint is now of type `GazePointAndDirection` (for (3))
5. Publishing the head position in the robot frame under `clm_ros_wrapper/head_position_rf`
6. Inside/outside checks using the width and height of the display screen and radii of free objects
7. Made target_detection publish the detected target without the Scene information by publishing a region code to see roughly what region it is (0 for screen, 1 for robot and 2 for outside)

Commit:
Moved the parameters for cf2wf transformation matrix to the parameter server. Accessible from the launch file

Commit:
Removed the double definiton of head position

Commit:
Corrected translation_vector_cf2wf

Commit:
New msg type Objects to get gaze points for both robot and screen. Made Scene an array of Objects.msg's

Commit:
Added robot-width and robot-height to the launch file. Edited scene_publisher to publish a scene with two groups of Objects

Commit:
Renamed Objects as ObjectsOnPlane. Added it to CMake file

Commit:
Added geometry_msgs to the dependencies for ObjectsOnPlane.msg

Commit:
Changed Objects to ObjectsOnPlane

Commit:
Created the FreeObject message for robot

Commit:
Renamed ObjectsOnPlane as Screen

Commit:
Edited scene_publisher, Scene.msg and CMakeLists for messages Screen and FreeObject

Commit:
Fixed typo in Scene.msg

Commit:
Added FreeObject to CMakeList. Included its header file in scene_publisher. Fixed typo in the push_back for loop of free objects in scene_publisher

Commit:
Short comment about inside/outside checkgit

Commit:
Removed reference points from FreeObject. Will use Point-Line distance formulae for free objects instead of Line-Plane intersection.

Commit:
Fixed the scene callback

Commit:
Removed reference points

Commit:
Created message type GazePointAndDirection

Commit:
Changed the publisher type to GazePointAndDirection. New name for the topic, clm_ros_wrapper/gaze_point_and_direction

Commit:
Corrected the name of the publisher

Commit:
Changed the type of gazepoint_callback

Commit:
Changed no_detection to target_no_detection

Commit:
Addded outside option for free objects

Commit:
Added the missing line for global variable num_objects to scene_callback. Changed num_closest_target to num_closest_object_on_screen

Commit:
Added the feature to find the closest free object using Line-Point distance formulae

Commit:
Outputting the new closer object chosen from objects on screen and free objects

Commit:
Changed the name of the subscribed topic

Commit:
Added inside/outside check for the screen

Commit:
Added in/out check for free objects based on info about their radii

Commit:
Fixed the seg fault because of scene_callback

Commit:
Changed free_object_radii array to a float variable. Corrected its value in the launch file

Commit:
Created publisher head_position_rf and regarding rotation matrix and translation vector in .launch file

Commit:
Added the feature for publishing region codes even when Scene is not published  (0 for screen, 1 for robot and 2 for outside)

alecive commented:
First thing I would to would be to move from a single element assignment in the launch file to lists. As you can see here http://wiki.ros.org/roscpp/Overview/Parameter%20Server#Retrieving_Lists , you can retrieve lists from the Parameter server, and you can use some tricks like the ones shown here http://answers.ros.org/question/55987/parameter-list-from-launch-file/ to set them in the launch file. In this way, you would move from your design, that is this: 

```
<param name="rotation_cf2wf_11" type="double" value="-0.897904"/>
<param name="rotation_cf2wf_12" type="double" value="0.439951"/>
<param name="rotation_cf2wf_13" type="double" value="-0.0145582"/>
<param name="rotation_cf2wf_21" type="double" value="0.067152"/>
<param name="rotation_cf2wf_22" type="double" value="0.104216"/>
<param name="rotation_cf2wf_23" type="double" value="-0.992285"/>
<param name="rotation_cf2wf_31" type="double" value="-0.43504"/>
<param name="rotation_cf2wf_32" type="double" value="-0.891954"/>
<param name="rotation_cf2wf_33" type="double" value="-0.12312"/>
```

to a more compact representation.

alecive committed:
Added newline at the end of msg files

alecive commented:
A part from this, feel free to merge! You can also do it later on (but remember to do it :stuck_out_tongue:  )

Commit:
Fixed rotation matrix from world frame to camera frame

Commit:
Merge branch 'develop' of https://github.com/sociallyassistiverobotics/clm_ros_wrapper into develop

Commit:
Fixed rotation matrix from world frame to camera frame

cmhuang committed:
Fix the translation vector from camera frame to world frame

Commit:
Removed the zero vector in wf2rf transformation. Changed robot_position to translation_wf2rf in parameter server. More comments in launch file.

TODO: measure the position of the world frame origin in the robot
frame, use lists instead of individual parameters
